product-service/handlers/getProducts.ts

```diff
@@ -0,0 +1,34 @@
+import { APIGatewayProxyHandler, APIGatewayProxyResult } from 'aws-lambda';
+import 'source-map-support/register';
+
+import { CORS_HEADERS } from '../common/constants';
+import repository from '../db/productRepository';
+
+export const getProducts: APIGatewayProxyHandler = async (event) => {
+  console.log('Lambda function getProducts has invoked');
+  console.log('event: ', event);
+
+  try {
+    const products = await repository.getProducts();
+
+    return {
+      statusCode: 200,
+      headers: CORS_HEADERS,
+      body: JSON.stringify(products),
+    };
+  } catch (e) {
+    return errorFetchProducts(e);
+  }
+};
+
+function errorFetchProducts(e: {
+  message: any;
+}): Promise<APIGatewayProxyResult> {
+  return Promise.resolve({
+    statusCode: 500,
+    headers: CORS_HEADERS,
+    body: JSON.stringify({
+      message: `error then get products: ${e.message}`,
```

Comment:
Looks like typo: "error **then** get..." -> should be "error **when** get..."

Reply:
Agree.